src/Core/Domain/Cart/QueryResult/CartInformation/CartShipping.php

```diff
@@ -0,0 +1,103 @@
+<?php
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Cart\QueryResult\CartInformation;
+
+/**
+ * Holds data for cart shipping information
+ */
+class CartShipping
+{
+    /**
+     * @var ?int
+     */
+    private $selectedCarrierId;
+
+    /**
+     * @var string
+     */
+    private $shippingPrice;
+
+    /**
+     * @var bool
+     */
+    private $freeShipping;
+
+    /**
+     * @var CartDeliveryOption[]
+     */
+    private $deliveryOptions;
+
+    /**
+     * @param string $shippingPrice
+     * @param bool $freeShipping
+     * @param CartDeliveryOption[] $deliveryOptions
+     * @param int|null $selectedCarrierId
+     */
+    public function __construct(
+        string $shippingPrice,
+        bool $freeShipping,
+        array $deliveryOptions,
+        ?int $selectedCarrierId
+    ) {
+        $this->shippingPrice = $shippingPrice;
+        $this->freeShipping = $freeShipping;
+        $this->deliveryOptions = $deliveryOptions;
```

Comment:
Why do you need delivery options here?

Reply:
You think i should load them on click?

Reply:
You want to get current cart shipping data, not something cart _could have_, as I understand.

Reply:
@matks, wdyt, is it worth? I'm currently trying to implement separate query for deliveryOptions, but its getting hard to render it dynamically, even tho it makes sense to get options separately, it seems to be loosing cohesion... Maybe the whole idea of CartInformation object to be returned to javascript is wrong?

Reply:
@sarjon Loading data before it is used can be a valid strategy it it serves an optimization purpose. If the data is already available at the time of the click, this makes it very quick.

If the code is readable and relevant (and it seems OK to have `delivery options` returned with `shipping costs`), if it does not amper drastically the response time and size and if this makes the UI easier to build, debug and customize, then yes let's do this ✅ 

I prefer we have a few too big Queries that return too much data rather than have multiple Queries which make the data scarce and hard to gather / process. Main target is simplicity and readability.